lib/soap.js

```diff
@@ -0,0 +1,274 @@
+            // This seems like a well-formed XML string, so parse it as such
+            const opts = {
+              explicitArray: false,
+              charkey: '#text',
+              mergeAttrs: true,
+              trim: true
+            };
+            try {
+              return mimecontent(value, 'text/xml').toObject(opts);
+            } catch (err) {}
+          }
+        }
+        return value;
+      });
+
+      const searchTerm = toRegex(vars.outcome_search_term);
+      const searchOutcome  = _.result(vars, 'outcome_on_match.trim', 'success').toLowerCase()
+      const searchPath     = _.result(vars, 'outcome_search_path.trim');
+      const reasonSelector = _.result(vars, 'reason_path.trim');
+      const priceSelector  = _.result(vars, 'price_path.trim');
+
+      // narrow the search scope
+      
+      const searchIn =
+        searchPath ?
+          // limit outcome_search_term to this path in the document
+          _.get(result, searchPath).toString()
+
+        :
+          // no search path was provided, so search the entire response body
+          body;
+
+      // look in all possible search scopes for the term
+      const found = searchIn != null ? searchIn.toLowerCase().match(searchTerm) : undefined;
+
+      // determine the outcome based on whether the search term was found
+      const outcome =
+        found ?
+          searchOutcome
+        :
+          inverseOutcome(searchOutcome);
+
+      const price = 
+        outcome === 'success' ?
+          _.get(result, priceSelector) : undefined;
+
+
+      // determine the reason based on the reason selector
+      const reasons =
+        reasonSelector ?
+          _.get(result, reasonSelector)
+        :
+          [];
+
+      // trim and comma delimit reasons
+      let reason = _(ensureArray(reasons))
+        .map(_.trim)
+        .compact()
+        .sort()
+        .join(', ');
+
+      // set the default reason, if necessary
+      if (!reason) { reason = _.result(vars, 'default_reason.trim') }
+
+      // build the event
+      const event = result;
+      event.outcome = outcome;
+      if (reason) { event.reason = reason; }
+      event.price = price || 0;
+
+      // return the event
+      return callback(null, event);
+    };
+
+    // calculate the timeout
+    const timeoutMs = (vars.timeout_seconds != null ? vars.timeout_seconds : 10) * 1000;
+
+    // call the function
+    fxn(args, handleResponse, {timeout: timeoutMs});
+  });
+};
+
+
+
+//
+// Validate ---------------------------------------------------------------
+//
+
+const validateFunction = function(vars) {
+  if (vars.function == null) { return 'Function is required'; }
+  if (!vars.function.match(/^[a-zA-Z0-9_.]+$/)) { return 'Function must have valid name'; }
+};
+
+
+const validateVersion = function(vars) {
+  const version = vars.version != null ? vars.version.trim() : undefined;
+  if (version == null) { return; }
```

Comment:
The previous line gives us some type guarantees, so we can shorten this
```suggestion
  if (!version) { return; }
```